library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImplInternal.java

```diff
@@ -1046,7 +1048,9 @@ private void doSomeWork() throws ExoPlaybackException, IOException {
       }
       if (!playbackInfo.isLoading
           && playbackInfo.totalBufferedDurationUs < 500_000
-          && isLoadingPossible()) {
+          && isLoadingPossible()
+          && stuckLoadingRetryCount++ >= 3) {
+        stuckLoadingRetryCount = 0;
```

Comment:
Once the player jumps past the gap it needs one more handler message (the load request one is pending here).  If we wanted to be more pedantic about it one could check the message queue was devoid of anything else (seek, etc.) that might clear the stuck-ness before triggering this error.

BTW, still working on the check in queuing samples for un-reported discontinuities.   In our experience that is the largest reason why this set of conditions occur in the render loop.

Reply:
This stuckLoading issue is no longer possible with GAP.  See my last commit b759e28fc 
The problem is `shouldContinueLoading()` is not using the correct method to compute buffered duration.

@ojw28 Could this be more of the stuck while loading is possible issues?